Tests/LibWasm/test-wasm.cpp

```diff
@@ -281,7 +281,13 @@ JS_DEFINE_NATIVE_FUNCTION(WebAssemblyModule::wasm_invoke)
             }
             break;
         case Wasm::ValueType::Kind::F32:
-            arguments.append(Wasm::Value(static_cast<float>(double_value)));
+            if (argument.is_object()) {
+                float float_value = 0;
+                memcpy(&float_value, &double_value, sizeof(float));
```

Comment:
That's.. interesting
What's happening again?

Reply:
Hmm yeah this part is a little weird haha. If the argument is an object, we know that it's a `Uint8Array` of size <= 8. It should've been copied into `double_value` earlier. But since the argument is a float, the `Uint8Array` will be of size 4, so I'm copying the first four bytes of the double into a float, which should be the value.

Maybe it's worth refactoring the method for passing arguments...

Reply:
yeah please either refactor, or at least leave a comment explaining the weirdness :)